fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -493,18 +521,29 @@ public boolean commitAndPublishTransaction(Database db, long transactionId,
     public void abortTransaction(Long dbId, String label, String reason) throws UserException {
         Preconditions.checkNotNull(label);
         Long transactionId = null;
-        writeLock();
+        readLock();
         try {
-            Map<String, Long> dbTxns = dbIdToTxnLabels.row(dbId);
-            if (dbTxns == null) {
+            Set<Long> existingTxns = dbIdToTxnLabels.get(dbId, label);
+            if (existingTxns == null || existingTxns.isEmpty()) {
                 throw new UserException("transaction not found, label=" + label);
             }
-            transactionId = dbTxns.get(label);
-            if (transactionId == null) {
+            // find PREPARE txn
```

Comment:
Please comment why find an prepare, if transaction state is not PREPARE, what can it be? Better to add check to the state. Why there can not be multiple PREPARE transactions.

Reply:
For one load label, there should be only one PREPARE txn. I will add comment here.